### ISSUE TITLE
Quickstart: `hub.Receive` not receiving "hello, world" 

### ISSUE THREAD

**Issue by atecce** [Azure/azure-event-hubs-go#101]
### Expected Behavior

seeing `hello, world!` print to `stdout` when running the Quickstart with my connection string

### Actual Behavior

~`amqp: receiver settlement mode "second" requested, received "first" from server`~

~fwiw i varied the credentials and the namespace and got other (more expected errors), so I'm confident it isn't something like mistyped names~ was using the connection string from the _namespace_ and not the event hub itself

simply iterates through each partition and hangs. errors are marked as `nil`

I know the messages are being sent because I can see the traffic in the portal

![Screen Shot 2019-04-11 at 2 54 49 PM](https://user-images.githubusercontent.com/10567112/55983728-c6508d00-5c69-11e9-948e-030b4c54c033.png)

### Environment
- OS: macOS 10.14.3
- Go version: 1.12.1
- Version of Library: 1.1.3

**Comment by atecce** [comment]
needed to `eventhub.ReceiveWithConsumerGroup(consumerGroup)`

**Comment by devigned** [comment]
I'm going to do a little more investigation on this one. You shouldn't run into the error above regardless of using `eventhub.ReceiveWithConsumerGroup(consumerGroup)`.

**Comment by atecce** [comment]
oh! well thank you!

just to be clear, the error message was a result of using the connection string from Event Hub Namespace instead of the Event Hub itself. 

after that a connection was successful, but `hub.Receive` didn't consume any messages.

to deal with that, I needed to supply an `eventhub.ReceiveOption` to `hub.Receive` in the sig like:

```go
hub.Receive(..., ..., ..., eventhub.ReceiveWithConsumerGroup(consumerGroup)
```

**Comment by devigned** [comment]
Thank you so much for the details, @atecce!

The namespace vs the hub conn string is something we should call out a bit more in the readme. The error message should provide some guidance to help in correcting the issue.

**Comment by atecce** [comment]
no problem at all!

**Comment by zhongyi-zhang** [comment]
Weird I hit `amqp: receiver settlement mode "second" requested, received "first" from server` when even using the connection string for the event hub itself.

**Comment by atecce** [comment]
looks like the test you cite there doesn't use a Consumer Group. would that explain it? are they required to subscribe and it doesn't fallback to a `$Default` or something?

**Comment by devigned** [comment]
The default consumer group if not specified is `$Default`.

Specifying the consumer group should not be related to the `amqp: receiver settlement mode "second" requested, received "first" from server`. The settlement mode error occurs when the message broker returns a different settlement mode than what the client requested. In PR #104, I should have the settlement issue resolved.

As part of that, I've also added an go example which is run in the test suite to ensure the "Hello World!" case is consistently executed.

**Comment by devigned** [comment]
I believe this issue is now resolved with #104. If you are still receiving the settlement mode error, please reopen this issue.
